### ISSUE TITLE
Data frame with character columns: unexpected results with R 3.5+

### ISSUE THREAD

**Issue by jmp75** [jmp75/rdotnet#97]
From a [comment in another issue](https://github.com/jmp75/rdotnet/issues/70#issuecomment-491258090):

In using R.NET 1.8.1 I have found an issue with getting values from character columns in a data frame. When I have a data frame with a character column, if I used `.AsDataFrame` to get a `DataFrame` object the character columns always contain empty values (`""`). Factor columns seem to work fine and accessing a character column directly with `.AsCharacter` also works fine. Below is an example. This was consistent for R 3.5.0, 3.5.3 and 3.6.0. In R 3.4.4 this works correctly. Could this be related to previous issues with character device?

```
        Dim engine As REngine
        Dim expression As SymbolicExpression

        REngine.SetEnvironmentVariables()
        engine = REngine.GetInstance()
        engine.Initialize()

        engine.Evaluate("x <- data.frame(c1 = c('a', 'b'), stringsAsFactors = FALSE)")
        expression = engine.GetSymbol("x")

        Console.WriteLine("Values as characters:")
        Console.WriteLine(expression.AsDataFrame(0)(0))
        Console.WriteLine(expression.AsDataFrame(0)(1))
        Console.WriteLine("*********************")

        engine.Evaluate("y <- data.frame(x = c('a', 'b'), stringsAsFactors = TRUE)")
        expression = engine.GetSymbol("y")

        Console.WriteLine("Values as factor:")
        Console.WriteLine(expression.AsDataFrame(0)(0))
        Console.WriteLine(expression.AsDataFrame(0)(1))
        Console.WriteLine("*********************")

        engine.Evaluate("c1 <- x$c1")
        expression = engine.GetSymbol("c1")

        Console.WriteLine("Values direct from column:")
        Console.WriteLine(expression.AsCharacter(0))
        Console.WriteLine(expression.AsCharacter(0))
        Console.WriteLine("*********************")
```
With output:
```
Values as characters:
'R.Net-Test.exe' (CLR v4.0.30319: R.Net-Test.exe): Loaded 'C:\WINDOWS\Microsoft.Net\assembly\GAC_MSIL\System.Numerics\v4.0_4.0.0.0__b77a5c561934e089\System.Numerics.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.


*********************
Values as factor:
a
b
*********************
Values direct from column:
a
a
*********************

```

**Comment by jmp75** [comment]
@dannyparsons I have looked at it but cannot reproduce any issue on R 3.4.4, 3.5.2, and a prerelease of R 3.6.  You can use the repro program at https://github.com/jmp75/rdotnet-support/blob/ec761b2be32f68df355ce4f1fe877c6a6217d693/usersupport/ReproUsers/Program.cs#L34  and see if you still encounter this issue on the machine you used.

**Comment by lrasmus** [comment]
@jmp75 - tried to reproduce but got the output below (which I think is expected).  Testing using:

- R.NET 1.8.1 (from NuGet)
- Console app targeting .NET Core 2.1
- R 3.5.1 x64

Curious if you were using 32-bit or 64-bit R in your testing?

Test program output (based on above example):

```
Values as characters:
a
b
*********************
Values as factor:
a
b
*********************
Values direct from column:
a
a
```

**Comment by dannyparsons** [comment]
@jmp75 @lrasmus thank you for looking into this. Strange that you do not have the same issue, but this also confirms what I thought that I didn't observe this issue when I first tested 1.8.0-alpha1 when it was initially released. I will test with your repo program and give feedback when I have more time next week. What other output would be useful for me to provide?

My original test was in a Windows Form App and below is the R version details, so its 32-bit R.

```
platform       i386-w64-mingw32          
arch           i386                      
os             mingw32                     
system         i386, mingw32             
status                                     
major          3                           
minor          5.3                         
year           2019                        
month          03                          
day            11                          
svn rev        76217                       
language       R                           
version.string R version 3.5.3 (2019-03-11)
nickname       Great Truth
```

**Comment by lrasmus** [comment]
Thanks @dannyparsons - let me re-test on 32-bit.  Very possibly something going on there.

**Comment by lrasmus** [comment]
Yes, I apparently missed part of the conversion code in the DynamicVector.cs file.  It looks like all other situations are handled.  This should address the issue: 
https://github.com/StatTag/rdotnet/commit/1d8a1101fe9a96659c0335ba3b47019aec720f45

**Comment by jmp75** [comment]
Thank you @lrasmus, your fix is in release 1.8.2

**Comment by dannyparsons** [comment]
Thanks for the fix! This is working well for me in 1.8.2.
